### PR TITLE
Update paths.hpp for v1/3

### PR DESCRIPTION
**Pull request by james-d-mitchell**
Currently missing support for `ReversiblePaths`.

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
Should be ready to go. The CI will fail until 

https://github.com/libsemigroups/libsemigroups/pull/587

is merged.

File: docs/source/data-structures/word-graph/paths/index.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Paths
+=====
+
+This page contains links to the documentationin ``libsemigroups_pybind11`` for finding and counting paths in a word graph. The relevant classes are:
+
+.. toctree::
+   :maxdepth: 1
+
+   paths
+   reversible-paths
```
Comment (Joseph-Edwards) [inline docs/source/data-structures/word-graph/paths/index.rst:18]: Does this file exist?

**Review by Joseph-Edwards — COMMENTED** [review]
Other than the non-existence of `reversible-paths.rst`, this looks good to me. Thanks @james-d-mitchell!

File: docs/source/data-structures/word-graph/paths/index.rst (hunk 1 above)
Comment (james-d-mitchell) [reply docs/source/data-structures/word-graph/paths/index.rst:18]: Argh, forgot to add it ⚡

**Comment by james-d-mitchell** [comment]
Thanks @Joseph-Edwards !

**Pull request merged**